api/v1alpha1/destination_types.go

```diff
@@ -49,6 +49,15 @@ type DestinationSpec struct {
 	//+kubebuilder:validation:Optional
 	StateStoreCoreFields `json:",inline"`
 	StateStoreRef        *StateStoreReference `json:"stateStoreRef,omitempty"`
+
+	// By default, Kratix will schedule works without labels to all destinations
+	// (for promise dependencies) or to a random destination (for resource
+	// requests). If StrictMatchLabels is true, Kratix will only schedule works
+	// to this destination if the labels match strictly (that is, an empty label
+	// set on the work won't be schedule to this destination, unless the
+	// destination label set is also empty).
+	// +kubebuilder:validation:Optional
+	StrictMatchLabels bool `json:"strictMatchLabels,omitempty"`
```

Comment:
> that is, an empty label set on the work won't be schedule to this destination, unless the destination label set is also empty

This is the only use case right? I think its hard to drive that understanding from the key name, I'd definitely think it means "only scheduling Work to this that have all my Destinations labels, not a subset". But I cannot think of a better name to describe what the comment highlights well. I think its a smell this behaviour from Kratix is weird... but as discussed yesterday we aren't going to change it yet. So I guess :lgtm: 🤷

Reply:
yeah, correct.

this is really "dontScheduleToThisDestinationsIfNoLabelsAreSet".

We _could_ make it _really_ strict matching, as in, only schedule to this destination if all `labels match` rather than a subset 🤔

Reply:
This comment is a little confusing, could it be clearer to talk about this in relation to only matching the DestinationSelectors since they are the only labels in question. Something like `matchDestinationSelector` could be fitting as we aren't being so strict as to match all of them.

Conversely, this could exist in the comment itself e.g. 

```
	// By default, Kratix will schedule works without labels to all destinations
	// (for promise dependencies) or to a random destination (for resource
	// requests). If StrictMatchLabels is true, Kratix will only schedule works
	// to this destination if destination matches the labels - reflecting the 
        // DestinationSelector configured on the Promise. An empty label
	// set on the work won't be schedule to this destination, unless the
	// destination label set is also empty
```

Reply:
@SaphMB are you suggesting we change the name of the field? From `StrictMatchLabels` to `MatchDestinationSelector`?

(i like the comment btw, will update)

Reply:
I'm very wary that there is a difference between what is clearer for me and what is clearer for the majority of users. That's what I was suggesting if that would actually be more useful for users who aren't me, but I think the updated comment would also suffice

On that note, I did make some typos there:
```
	// By default, Kratix will schedule works without labels to all destinations
	// (for promise dependencies) or to a random destination (for resource
	// requests). If StrictMatchLabels is true, Kratix will only schedule works
	// to this destination if the destination matches the labels - reflecting the 
        // DestinationSelector configured on the Promise. An empty label
	// set on the work won't be scheduled to this destination, unless the
	// destination label set is also empty
```

Reply:
I think I like `MatchDestinationSelectors` better tbf -- i was just double checking my understanding 😛

Reply:
I'll change the comment-documentation to say 

> If StrictMatchLabels is true, Kratix will only schedule works to this destination if the destination can be matched by the defined DestinationSelectors